Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ContextMenu.cpp

```diff
@@ -43,18 +43,26 @@ namespace EMStudio
 
         QMenu* nodeGroupMenu = new QMenu(tr("Assign To Node Group"), menu);
 
+        int validGroupCount = 0;
         for (size_t i = 0; i < numNodeGroups; ++i)
         {
             EMotionFX::AnimGraphNodeGroup* nodeGroup = animGraph->GetNodeGroup(i);
-
-            AZ::Color nodeGroupColor;
-            nodeGroupColor.FromU32(nodeGroup->GetColor());
-            QAction* nodeGroupAction = nodeGroupMenu->addAction(QIcon(new SolidColorIconEngine(nodeGroupColor)), nodeGroup->GetName());
-            nodeGroupAction->setData(qulonglong(i + 1)); // index of the menu added, not used
-            connect(nodeGroupAction, &QAction::triggered, this, &BlendGraphWidget::AssignSelectedNodesToGroup);
+            auto* activeGraph = GetActiveGraph();
+
+            if ((!nodeGroup->ParentNodeId().IsValid() && !activeGraph->GetAnimGraphModel().GetFocus().isValid()) ||
+                  nodeGroup->ParentNodeId() == activeGraph->GetAnimGraphModel().GetFocus().data(AnimGraphModel::ROLE_ID).value<EMotionFX::AnimGraphNodeId>()) {
+                AZ::Color nodeGroupColor;
+                nodeGroupColor.FromU32(nodeGroup->GetColor());
+                QAction* nodeGroupAction = nodeGroupMenu->addAction(QIcon(new SolidColorIconEngine(nodeGroupColor)), nodeGroup->GetName());
+                nodeGroupAction->setData(qulonglong(i + 1)); // index of the menu added, not used
+                connect(nodeGroupAction, &QAction::triggered, this, &BlendGraphWidget::AssignSelectedNodesToGroup);
+                validGroupCount++;
+            }
         }
 
-        menu->addMenu(nodeGroupMenu);
+        if (validGroupCount > 0) {
+            menu->addMenu(nodeGroupMenu);
+        }
```

Comment:
standards: Please move opening braces (`{`) to a new line (both `if` conditions)

suggestion: Again a comment here might not hurt for future maintainers (for the conditional) so it's a bit less cryptic to understand what is happening at a high-level (or move this to a named function that can have a clearer name)

Reply:
Done